### PR TITLE
Forman xxx fix subset from large images

### PR DESCRIPTION
**Pull request by forman**
[Description of PR]

Checklist:

* [x] Add unit tests and/or doctests in docstrings
* [ ] ~Add docstrings and API docs for any new/modified user-facing classes and functions~
* [x] New/modified features documented in `docs/source/*`
* [x] Changes documented in `docs/CHANGES.md`
* [x] AppVeyor and Travis CI passes
* [x] Test coverage remains or increases (target 100%)

### CODE REVIEW

File: xcube/core/geocoding.py [hunk 1]
```diff
@@ -419,7 +422,16 @@ def gu_compute_ij_bboxes(x_image: np.ndarray,
             ij_bboxes[k, 3] = j_max
 
 
-def _get_dataset_xy_names(dataset: xr.Dataset, xy_names: Tuple[str, str] = None) -> Tuple[str, str]:
+def get_xy_var(dataset: xr.Dataset, name: str) -> xr.DataArray:
```
Comment (TonioF) [inline xcube/core/geocoding.py:425]: I'd even prefer the new name get_cood_var, but get_xy_var is also fine.

**Review by TonioF — COMMENTED** [review]
So ... the correct solution would come with the integration of the resampling?

**Comment by forman** [comment]
> So ... the correct solution would come with the integration of the resampling?

Well this one is also right, but with the resampling comes a cleaner solution.

**Review by TonioF — APPROVED** [review]
I

**Pull request merged**